### PR TITLE
Add overrides to RequireManagement for more flexible configuration

### PR DESCRIPTION
**Pull request by stevie400**
This should allow for more flexible configuration of this plugin. Our use case, is not requiring management and allowing versions for all `com.hubspot*` stuff.

@jhaber

### CODE REVIEW

File: src/main/java/com/hubspot/maven/plugins/dependency/management/RequireManagementOverride.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  public void setAllowVersions(boolean allowVersions) {
+    this.allowVersions = allowVersions;
+  }
+
+  public void setAllowExclusions(boolean allowExclusions) {
+    this.allowExclusions = allowExclusions;
+  }
+
+  public RequireManagmentConfig toRequireManagementConfig(final RequireManagement requireManagement) {
+    return new RequireManagmentConfig() {
```
Comment (stevie400) [inline src/main/java/com/hubspot/maven/plugins/dependency/management/RequireManagementOverride.java:38]: Should I memoize this?

File: README.md [hunk 2]
```diff
@@ -61,9 +61,14 @@ run as a standalone plugin or as a rule with the maven-enforcer-plugin.
             <plugins>true</plugins>
             <allowVersions>false</allowVersions>
             <allowExclusions>false</allowExclusions>
-            <exceptions>
-              <exception>com.example:*</exception>
-            </exceptions>
+            <overrides>
+              <override>
+                <exceptions>
```
Comment (stevie400) [inline README.md]: Any thoughts on this naming?

Comment (jhaber) [reply README.md]: Maybe `<patterns>`+`<pattern>`?

File: src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java [hunk 3]
```diff
@@ -117,18 +119,19 @@ private boolean checkPluginManagement() {
...
       key = key.substring(0, key.indexOf(':', key.indexOf(':') + 1));
     }
-
-    for (String exception : requireManagement.getExceptions()) {
-      if (SelectorUtils.match(exception, key)) {
-        return true;
+    for (RequireManagementOverride override : requireManagement.getOverrides()) {
+      for (String exception : override.getExceptions()) {
+        if (SelectorUtils.match(exception, key)) {
+          System.out.println("Using override:" + override);
```
Comment (jhaber) [inline src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java]: ^

Comment (stevie400) [reply src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java]: 😳

File: src/main/java/com/hubspot/maven/plugins/dependency/management/RequireManagementOverride.java (hunk 1 above)
Comment (jhaber) [reply src/main/java/com/hubspot/maven/plugins/dependency/management/RequireManagementOverride.java:38]: I wouldn't worry about it

File: README.md (hunk 2 above)
Comment (stevie400) [reply README.md]: 👍

**Comment by jhaber** [comment]
lgtm

**Pull request merged**